src/seedu/addressbook/TextUi.java

```diff
@@ -0,0 +1,327 @@
+     * editing and proof reading. Such messages are considered part of the UI
+     * and may be subjected to review by UI experts or technical writers. Note
+     * that Some of the strings below include '%1$s' etc to mark the locations
+     * at which java String.format(...) method can insert values.
+     * ====================================================================
+     */
+
+    public static final String MESSAGE_ADD_PERSON_SUCCESS = "New person added: %1$s";
+    public static final String MESSAGE_ADDRESSBOOK_CLEARED = "Address book has been cleared!";
+    public static final String MESSAGE_CHANGES_SAVED_TO_STORAGE_FILE = "Changes saved to storage file.";
+    public static final String MESSAGE_COMMAND_HELP = "%1$s: %2$s";
+    public static final String MESSAGE_COMMAND_HELP_PARAMETERS = "\tParameters: %1$s";
+    public static final String MESSAGE_COMMAND_HELP_EXAMPLE = "\tExample: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+    public static final String MESSAGE_ERROR_MISSING_STORAGE_FILE = "Storage file missing: %1$s";
+    public static final String MESSAGE_ERROR_READING_FROM_FILE = "Unexpected error: unable to read from file: %1$s";
+    public static final String MESSAGE_ERROR_WRITING_TO_FILE = "Unexpected error: unable to write to file: %1$s";
+    public static final String MESSAGE_INDEXED_LIST_ITEM = "\t%1$d. %2$s";
+    public static final String MESSAGE_GOODBYE = "Exiting Address Book... Good bye!";
+    public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format: %1$s " + LS + "%2$s";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_INVALID_STORAGE_FILE_CONTENT = "Storage file has invalid content";;
+    public static final String MESSAGE_INVALID_STORAGE_FILE_PATH = "Storage file should be valid and end with \".txt\"";
+    public static final String MESSAGE_PERSON_NOT_IN_ADDRESSBOOK = "Person could not be found in address book";
+    public static final String MESSAGE_PERSONS_FOUND_OVERVIEW = "%1$d persons found!";
+    public static final String MESSAGE_WELCOME = "Welcome to your Address Book!";
+    public static final String MESSAGE_USING_STORAGE_FILE = "Using storage file : %1$s";
+    public static final String MESSAGE_VIEW_PERSON_DETAILS = "Viewing person: %1$s";
+
+    public static final String COMMAND_ADD_DESC = "Adds a person to the address book.";
+    public static final String COMMAND_ADD_PARAMETERS = "NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...";
+    public static final String COMMAND_ADD_EXAMPLE = Command.ADD.commandWord + 
+            " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owes_money";
```

Comment:
I was thinking of it. Just the part where we upgrade commands into first class objects, but we can ignore invokers etc. I would like to do it, it will shorten the code and reduce complexity by quite a bit.